src/java.base/share/classes/java/util/Random.java

```diff
@@ -82,6 +83,159 @@
         equidistribution = 0
 )
 public class Random implements RandomGenerator, java.io.Serializable {
+
+    /**
+     * Class used to wrap a {@link java.util.random.RandomGenerator} to
+     * {@link java.util.Random}.
+     */
+
+    @SuppressWarnings("serial")
+    private static final class RandomWrapper extends Random implements RandomGenerator {
+        private final RandomGenerator generator;
+
+        private RandomWrapper(RandomGenerator randomToWrap) {
+            super(null);
+            this.generator = Objects.requireNonNull(randomToWrap);
```

Comment:
Can remove `requireNonNull` here; see other comments. At your option, add a comment saying that `randomToWrap` must not be null. This is easy to verify since there's only one caller elsewhere in this file.